pallets/api/src/fungibles/tests.rs

```diff
@@ -9,165 +9,212 @@ use frame_support::{
+		assert_ok!(Fungibles::decrease_allowance(signed(ALICE), id, BOB, 0));
+		assert_eq!(Assets::allowance(id, &ALICE, &BOB), value);
 		// Decrease allowance successfully.
-		assert_ok!(Fungibles::decrease_allowance(signed(ALICE), ASSET, BOB, amount / 2 - 1 * UNIT));
-		assert_eq!(Assets::allowance(ASSET, &ALICE, &BOB), amount / 2 + 1 * UNIT);
+		assert_ok!(Fungibles::decrease_allowance(signed(ALICE), id, BOB, value / 2));
+		assert_eq!(Assets::allowance(id, &ALICE, &BOB), value / 2);
+		System::assert_last_event(Event::Approval { id, owner, spender, value: value / 2 }.into());
 		// Saturating if current allowance is decreased more than the owner balance.
-		assert_ok!(Fungibles::decrease_allowance(signed(ALICE), ASSET, BOB, amount));
-		assert_eq!(Assets::allowance(ASSET, &ALICE, &BOB), 0);
+		assert_ok!(Fungibles::decrease_allowance(signed(ALICE), id, BOB, value));
+		assert_eq!(Assets::allowance(id, &ALICE, &BOB), 0);
+		System::assert_last_event(Event::Approval { id, owner, spender, value: 0 }.into());
 	});
 }
 
 #[test]
 fn create_works() {
 	new_test_ext().execute_with(|| {
-		assert!(!Assets::asset_exists(ASSET));
-		assert_ok!(Fungibles::create(signed(ALICE), ASSET, ALICE, 100));
-		assert!(Assets::asset_exists(ASSET));
+		let id = ASSET;
+
+		assert!(!Assets::asset_exists(id));
+		assert_ok!(Fungibles::create(signed(ALICE), id, ALICE, 100));
+		assert!(Assets::asset_exists(id));
 	});
 }
 
 #[test]
 fn start_destroy_works() {
 	new_test_ext().execute_with(|| {
-		create_asset(ALICE, ASSET);
-		assert_ok!(Fungibles::start_destroy(signed(ALICE), ASSET));
+		let id = ASSET;
+
+		create_asset(ALICE, id);
+		assert_ok!(Fungibles::start_destroy(signed(ALICE), id));
 	});
 }
 
 #[test]
 fn set_metadata_works() {
 	new_test_ext().execute_with(|| {
+		let id = ASSET;
 		let name = vec![42];
 		let symbol = vec![42];
 		let decimals = 42;
-		create_asset(ALICE, ASSET);
+
+		create_asset(ALICE, id);
 		assert_ok!(Fungibles::set_metadata(
 			signed(ALICE),
-			ASSET,
+			id,
 			name.clone(),
 			symbol.clone(),
 			decimals
 		));
-		assert_eq!(Assets::name(ASSET), name);
-		assert_eq!(Assets::symbol(ASSET), symbol);
-		assert_eq!(Assets::decimals(ASSET), decimals);
+		assert_eq!(Assets::name(id), name);
+		assert_eq!(Assets::symbol(id), symbol);
+		assert_eq!(Assets::decimals(id), decimals);
 	});
 }
 
 #[test]
 fn clear_metadata_works() {
 	new_test_ext().execute_with(|| {
-		let name = vec![42];
-		let symbol = vec![42];
-		let decimals = 42;
-		create_asset_and_set_metadata(ALICE, ASSET, name, symbol, decimals);
-		assert_ok!(Fungibles::clear_metadata(signed(ALICE), ASSET));
-		assert_eq!(Assets::name(ASSET), Vec::<u8>::new());
-		assert_eq!(Assets::symbol(ASSET), Vec::<u8>::new());
-		assert_eq!(Assets::decimals(ASSET), 0u8);
+		let id = ASSET;
+
+		create_asset_and_set_metadata(ALICE, id, vec![42], vec![42], 42);
+		assert_ok!(Fungibles::clear_metadata(signed(ALICE), id));
+		assert_eq!(Assets::name(id), Vec::<u8>::new());
```

Comment:
For future, could probably just check if it is empty - less verbose.

Reply:
I like it, thanks for pointing it out!

see [refactor: fungible pallet](https://github.com/r0gue-io/pop-node/pull/153/commits/3b3ae9fcf4de6115bc5a2ecb9a92a2ca2e0d0068)